frontend/src/service-worker.js

```diff
@@ -1,15 +1,89 @@
-import {serviceWorkerFetchListener} from "sync-message";
+/* eslint-disable */
+// Otherwise webpack can fail silently
+// https://github.com/facebook/create-react-app/issues/8014
+// Reference for service-worker.js: https://developers.google.com/web/tools/workbox/modules
 
-console.log(self.__WB_MANIFEST);
+import {serviceWorkerFetchListener} from 'sync-message'; // Do not remove
+const fetchListener = serviceWorkerFetchListener(); // Do not remove
+addEventListener('fetch', fetchListener); // Do not remove
 
-const fetchListener = serviceWorkerFetchListener();
+import {clientsClaim} from 'workbox-core';
+import {ExpirationPlugin} from 'workbox-expiration';
+import {precacheAndRoute, createHandlerBoundToURL} from 'workbox-precaching';
+import {registerRoute} from 'workbox-routing';
+import {NetworkFirst, StaleWhileRevalidate} from 'workbox-strategies';
 
-addEventListener('fetch', fetchListener);
+clientsClaim();
 
-addEventListener('install', function (e) {
-  e.waitUntil(self.skipWaiting());
-});
+if (process.env.NODE_ENV === 'production') {
+  // Precache all of the assets generated by your build process.
+  // Their URLs are injected into the manifest variable below.
+  // This variable must be present somewhere in your service worker file,
+  // even if you decide not to use precaching. See https://cra.link/PWA
+  precacheAndRoute(self.__WB_MANIFEST);
+}
+
+// Set up App Shell-style routing, so that all navigation requests
+// are fulfilled with your index.html shell. Learn more at
+// https://developers.google.com/web/fundamentals/architecture/app-shell
+const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$');
+registerRoute(
+  // Return false to exempt requests from being fulfilled by index.html.
```

Comment:
I don't understand what this is doing, do you?

Reply:
I looked at the comments and the link but still wasn't quite sure I understood the code.

Reply:
OK cool, looks like it can all go inside of `"production"`. Tested both `npm run start` (to make sure there were no regressions) and `npm run pwa`.